### ISSUE TITLE
Using fetch from PageEvent results in the wrong content being cached in certain circumstances

### ISSUE THREAD

**Issue by tedsteen** [sveltejs/kit#7514]
### Describe the bug

When fetching from f.ex a graphql endpoint, most of the time the URL is the same but the body (query) is different. This causes the fetch provided by the PageLoad event to use the wrong cached data during hydration.

### Reproduction

Here's an example and description of the problem.
https://github.com/tedsteen/reproducables/pull/1

### System Info

```Shell
System:
    OS: macOS 13.0
    CPU: (10) arm64 Apple M1 Pro
    Memory: 1.08 GB / 32.00 GB
    Shell: 5.9 - /opt/homebrew/bin/zsh
  Binaries:
    Node: 18.11.0 - /opt/homebrew/bin/node
    npm: 8.19.2 - /opt/homebrew/bin/npm
  Browsers:
    Brave Browser: 107.1.45.116
    Safari: 16.1
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.87
    @sveltejs/kit: next => 1.0.0-next.535
    svelte: ^3.44.0 => 3.52.0
    vite: ^3.1.0 => 3.2.2
```

### Severity

serious, but I can work around it

### Additional Information

I fixed this by creating a hash of the body sent to fetch and add it as a query parameter. Not sure if I'm missing something or this is a viable way to fix this.

**Comment by dummdidumm** [comment]
Hashing the body is probably what we need to do. Omitting POST request from the hydration behavior would be disastrous for anyone using GraphQL etc which do POST requests all the time.

**Comment by tedsteen** [comment]
Sorry guys but the issue was not fixed by #7531 😬 
I have updated [my reproducible](https://github.com/tedsteen/reproducables/pull/1) to kit 1.0.0-next.547 and it's still showing the same behaviour as before (see my description in the linked PR).
